### PR TITLE
Source Google Analytics (GA4): Fix Pagination and add Filters

### PR DESCRIPTION
**Pull request by mpetrykin**
## What
1. Fixed pagination;
2. Added filters dimensionFilter and metricFilter to the request body.

## How
1. The GA4 runReport hasn't **offset** and **limit** fields (you can see that in [official documentation](https://developers.google.com/analytics/devguides/reporting/data/v1/rest/v1beta/RunReportResponse)). Added a field **next_page_offset** to config and handling this decide the problem;
2. Change unit tests for next_page_token due to absent **offset** and **limit** fields in response.
3. Added filters structure (filters [official documentation](https://developers.google.com/analytics/devguides/reporting/data/v1/rest/v1beta/FilterExpression)) to **custom_reports_schema.json** for validation and handle this fields in **instantiate_report_class** method.

### CODE REVIEW

**Comment by marcosmarxm** [comment]
@mpetrykin can you check your commits? There is a user which didn't sign the CLA.

File: airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/defaults/custom_reports_schema.json [hunk 1]
```diff
@@ -21,6 +21,23 @@
...
+        "type": ["integer", "string"],
+        "minimum": 0,
+        "pattern": "^[0-9]+$"
+      },
+      "limit": {
+        "type": ["integer", "string"],
+        "minimum": 0,
+        "maximum": 250000,
+        "pattern": "^([0-9]|[1-9][0-9]{1,4}|1[0-9]{5}|2[0-4][0-9]{4}|250000)$"
+      },
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/defaults/custom_reports_schema.json:40]: Why are you adding this metadata into the schema of records?

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
@mpetrykin this is quite a big change. Please open an issue and describe clearly what are the problems you're trying to solve. Would be better to slit the change into 2 PRs though to simpler changes.

**Comment by marcosmarxm** [comment]
> ## What
> 1. Fixed pagination;
> 2. Added filters dimensionFilter and metricFilter to the request body.

Please first open an issue to discuss what you're intending to fix.

**Comment by mpetrykin** [comment]
Ok, I will close this PR and open 2 another with separated changes.
@marcosmarxm Could I ask you to help me, how to attach issue to PR?(
I created [issue](https://github.com/airbytehq/airbyte/issues/26066), but idk how to make the same number of PR with issue

**Comment by marcosmarxm** [comment]
> Ok, I will close this PR and open 2 another with separated changes. @marcosmarxm Could I ask you to help me, how to attach issue to PR?( I created [issue](https://github.com/airbytehq/airbyte/issues/26066), but idk how to make the same number of PR with issue

Break the issue into 2.

**Pull request closed without merging**